app.rb

```diff
@@ -0,0 +1,133 @@
+require './person'
+require './student'
+require './teacher'
+require './classroom'
+require './book'
+require './rental'
+
+# App class for managing the school library application
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+    @classroom = Classroom.new('Grade 10')
+  end
+
+  # Lists all books in the library
+  def list_all_books
+    if @books.empty?
+      puts 'Oops! the library is empty, please add books'
+      return
+    end
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  # Lists all registered people (students and teachers)
+  def list_all_people
+    if @people.empty?
+      puts 'Oops! no people registered, please add people'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  # Creates a person (student or teacher) based on user input
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher(2)?....'
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input! Please type 1 or 2'
+    end
+  end
+
+  # Creates a student based on user input
+  def create_student
+    print 'Name:'
+    name = gets.chomp
+
+    print 'Age:'
+    age = gets.chomp.to_i
+
+    print 'Has Parent permission? [Y/N]:'
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(age, @classroom, name, parent_permission: parent_permission == 'y')
+    @people << student
+    puts 'Student created successfully'
+  end
```

Comment:
- [ ] KIndly make sure when creating the new student the `age` and `classroom` are being stored in the format as it was defined in the `Student` class.

     - In your `student` class the input format is like this `(classroom, age, ...)` but when you are passing the inputs provided by the user you did the opposite `(age, @classroom, ...)` and this is cause a problem when user wants to list all people, instead of getting the exact `student` age, the `classroom` object shows up. Kindly fix that.

      - ### YOUR LIBRARY UI BEHAVIOUR
      
   ![image](https://github.com/raihan2bd/school-library/assets/87197412/9d2411c9-3f52-4935-a50c-db03c761cde1)


```suggestion
  def create_student
    print 'Name:'
    name = gets.chomp

    print 'Age:'
    age = gets.chomp.to_i

    print 'Has Parent permission? [Y/N]:'
    parent_permission = gets.chomp.downcase

    student = Student.new(@classroom, age,  name, parent_permission: parent_permission == 'y')
    @people << student
    puts 'Student created successfully'
  end
```

Reply:
Hi @levy002 ✋,  Thanks a lot for your valuable suggestions. I will implement your suggestions before the next review 👍